### PR TITLE
feat: `Alert` component

### PR DESCRIPTION
**Pull request by jordmccord**
Adds `Alert` component, see: https://www.figma.com/design/4FFYTLWJ2hQpj36JplQQUw/UW-Web-UI?node-id=5367-21799

### CODE REVIEW

**Review by robphoenix — CHANGES_REQUESTED** [review]
This great @jordmccord, thankyou. 🙇🏻 

There's a couple of things we need to speak with Noelia about; the colorScheme naming (see my comment in Figma), and the icon buttons.

File: packages/web-ui/src/Alert/AlertLink.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    outline: 'none',
+    borderRadius: px(4),
+    boxShadow: '0 0 0 2px var(--alert-focus-color)',
+  },
+  '&:hover': {
+    textDecoration: 'none',
+  },
+});
+
+AlertLink.displayName = componentName;
```
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/AlertLink.tsx]: What is this syntax `[`${COLORSCHEME_SELECTORS.info} &`]: {` supposed to do? I've removed the styling from the parent and now the colour change isn't applied?

File: packages/web-ui/src/Alert/Alert.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    color: 'var(--alert-text-color)',
+  },
+  '> :where(a)': {
+    color: 'var(--alert-link-color)',
+  },
+  '> :where(button)': {
+    color: 'var(--alert-icon-color)',
+    backgroundColor: 'transparent',
+    border: 'none',
+    cursor: 'pointer',
```
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx:87]: Ok cool, yeah I had questioned no hover states. Can I ask why no pointer? I know the default browser button doesn't use a pointer...

File: packages/web-ui/src/Alert/Alert.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+import { AlertLink } from './AlertLink';
+
+const componentName = 'Alert';
+const componentClassName = withGlobalPrefix(componentName);
+
+const classNames = {
+  direction: {
+    row: withGlobalPrefix('direction-row'),
+    column: withGlobalPrefix('direction-column'),
+  },
```
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx]: Check my implementation but I've used data attrs over class names. The data orientation comes from the radix radio group so I've had to add it to our data attrs

File: packages/web-ui/src/Alert/AlertText.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * An `AlertText` is a component that is used to display the text of an `Alert`.
+ *
+ * > This component does not need to be wrapped in a `ThemeProvider` and can be used standalone with other component libraries.
+ */
+export const AlertText = styled('span')({
+  fontSize: pxToRem(16),
+  lineHeight: pxToRem(24),
+  color: 'var(--alert-text-color)',
+});
```
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/AlertText.tsx]: I have implemented but the ``` [`${COLORSCHEME_SELECTORS.success} &`]: { ``` doesn't work as I think you intend it to when removing `'--alert-link-color': colors.green700,` from the parent.

File: packages/web-ui/src/utils/data-attributes.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ export const DATA_ATTRIBUTES = {
   customColor: withDataPrefix('custom-color'),
   colorscheme: withDataPrefix('colorscheme'),
   inverted: withDataPrefix('inverted'),
+  orientation: withDataPrefix('orientation'),
```
Comment (robphoenix) [inline packages/web-ui/src/utils/data-attributes.ts:12]: 👍🏻

File: packages/web-ui/src/Alert/Alert.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (robphoenix) [reply packages/web-ui/src/Alert/Alert.tsx:73]: - https://adamsilver.io/blog/buttons-shouldnt-have-a-hand-cursor/
- https://adamsilver.io/blog/buttons-shouldnt-have-a-hand-cursor-part-2/

File: packages/web-ui/src/Alert/Alert.tsx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        ref={ref}
+        className={clsx(componentClassName, className)}
+        role="alert" // Adding role for dynamic alerts
+        aria-live="assertive" // Making it announced immediately
+        aria-atomic="true" // Ensuring the entire alert is read as a whole
+        {...dataAttributeProps}
+        {...props}
+      >
+        <AlertIcon colorScheme={colorScheme} />
+        <div className={classNames.content}>
```
Comment (robphoenix) [inline packages/web-ui/src/Alert/Alert.tsx:158]: I'm kinda unsure about this approach, though I don't know if I can fully articulate why. I would rather use DS components, like `Flex` than create multiple generic classnames which aren't related to some kind of state/variant or prop. I'd worry that a class such as `uwu-content` or similar becomes arbitrary and used across different components without any reason behind it. I also feel we should be using DS components as much as possible.

Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx:158]: Yeah fair point, I'll make the change 👍

Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx:158]: What about the unstyled button?

File: packages/web-ui/src/Alert/Alert.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx:46]: Hmm, interesting. I guess that there's an argument for what is semantically a button but presentationally not (a traditional button). As long as we have either a pointer or hover state to indicate that it's interactive I'm happy, but you're right it's a design decision.

[the remark this answers is not in the document]
Comment (robphoenix) [reply packages/web-ui/src/Alert/Alert.tsx:46]: Noelia wants the button cursors changed to pointer.

File: packages/web-ui/src/Alert/Alert.props.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { ComponentPropsWithoutRef } from 'react';
+
+export interface AlertProps extends ComponentPropsWithoutRef<'div'> {
+  /**
+   * Sets the colour scheme.
+   * @default info
+   */
+  colorScheme?: 'info' | 'success' | 'warning' | 'error';
```
[the remark this answers is not in the document]
Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.props.ts:8]: Updated to presentational values based on @noeliacabane's comment on Figma https://www.figma.com/design/4FFYTLWJ2hQpj36JplQQUw?node-id=9834-23189#846905338

**Review by robphoenix — CHANGES_REQUESTED** [review]

File: packages/web-ui/src/Alert/Alert.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      boxShadow: '0 0 0 2px var(--alert-focus-color)',
+    },
+  },
+  [COLORSCHEME_SELECTORS.cyan]: {
+    '--alert-text-color': colors.cyan900,
+    '--alert-background-color': colors.cyan50,
+    '--alert-icon-color': colors.cyan700,
+    '--alert-border-color': colors.cyan500,
+    '--alert-focus-color': colors.cyan700,
+    '--alert-button-hover-color': colors.cyan900,
```
Comment (robphoenix) [inline packages/web-ui/src/Alert/Alert.tsx:62]:
```suggestion
    '--alert-button-color-hover': colors.cyan900,
```

Comment (robphoenix) [reply packages/web-ui/src/Alert/Alert.tsx:62]: `--{component}-{element}-{property}-{modifier}`

File: packages/web-ui/src/Alert/Alert.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+      alignSelf: 'flex-start',
+    },
+  },
+  '> :where(svg, [data-icon])': {
+    color: 'var(--alert-icon-color)',
+  },
+  '> :where(button, a)': {
+    color: 'var(--alert-icon-color)',
+    alignSelf: 'center',
+    '&:hover': {
```
Comment (robphoenix) [inline packages/web-ui/src/Alert/Alert.tsx:47]:
```suggestion
     '@media (hover: hover)': {
      ':where(:hover)': {
```

File: packages/web-ui/src/Alert/Alert.tsx [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+      color: 'var(--alert-button-hover-color)',
+    },
+    '&:focus-visible': {
```
Comment (robphoenix) [inline packages/web-ui/src/Alert/Alert.tsx:50]:
```suggestion
  ':where(:focus-visible)': {
```

File: packages/web-ui/src/Alert/Alert.tsx (hunk 9 above)
Comment (robphoenix) [reply packages/web-ui/src/Alert/Alert.tsx:47]: fyi: https://jackdomleo.dev/blog/2020/hover-css-media-query

Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx:47]: I mean that's a step too far, I'll have read though 😂

Comment (jordmccord) [reply packages/web-ui/src/Alert/Alert.tsx]: In the example that's a neat way to show states by default for non-hover devices. Do we really need that here? I'm all for consistency but this seems overkill?

Comment (robphoenix) [reply packages/web-ui/src/Alert/Alert.tsx]: sure, probably is 👍🏻

File: packages/web-ui/src/Alert/Alert.docs.mdx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+## Advanced usage
+
+You can use the elements of a `Alert` separately for advanced usage for example changing the alert text:
+
+```tsx
+import { Alert, AlertTitle, AlertText, AlertLink, Strong } from '@utilitywarehouse/web-ui';
+
+const YourComponent = () => (
+  <Alert colorScheme="error">
```
Comment (robphoenix) [inline packages/web-ui/src/Alert/Alert.docs.mdx:36]:
```suggestion
  <Alert colorScheme="red">
```

**Review by robphoenix — APPROVED** [review]

**Comment by robphoenix** [comment]
👏🏻 👏🏻 👏🏻 🚢  😄

**Pull request merged**